addons/sale/models/sale.py

```diff
@@ -1390,10 +1390,9 @@ def _compute_untaxed_amount_to_invoice(self):
                 # amount and not zero. Since we compute untaxed amount, we can use directly the price
                 # reduce (to include discount) without using `compute_all()` method on taxes.
                 price_subtotal = 0.0
-                if line.product_id.invoice_policy == 'delivery':
-                    price_subtotal = line.price_reduce * line.qty_delivered
-                else:
-                    price_subtotal = line.price_reduce * line.product_uom_qty
+                umo_qty_to_consider = line.qty_delivered if line.product_id.invoice_policy == 'delivery' else line.product_uom_qty
+                price_reduce = line.price_unit * (1 - (line.discount or 0.0) / 100.0)
```

Comment:
What about the `line.price_reduce` 6 lines below? this one is correct and does not suffer from rounding issues ?

Shouldn't it be also replaced by the `price_reduce` var ?

Reply:
Yep, you're right!
I added a test for this case
(I also realized that in such a case, the considered quantity wasn't correct)